### ISSUE TITLE
netspeed unusable on vertical panels

### ISSUE THREAD

**Issue by pF-arQon** [mate-desktop/mate-panel#1411]
#### Expected behaviour

netspeed accurately displays network activity

#### Actual behaviour

netspeed displays what is effectively random numbers
![netspeed-oof](https://github.com/mate-desktop/mate-panel/assets/43844448/47443bba-f46b-4df7-804c-1fa38b3114f6)

Is that 5 megabits? 85? 965? Your guess is as good as mine.  :)

#### Steps to reproduce the behaviour

Create a vertical panel, add the netspeed applet to it. 

#### MATE general version

1.26

#### Linux Distribution

Ubuntu 22.04

#### Link to bugreport of your Distribution (requirement)

https://bugs.launchpad.net/ubuntu/+source/mate-applets/+bug/1841854 , which contains fixes for both this and the icon positioning bug

**Comment by lukefromdc** [comment]
I just tested a bunch of left panel widths, and found that this applet simply does not manage space well when confined horizontally. The minimum width necessary to allow the applet to be useful would be the amount needed to accomodate the decimal places though, it's never going to be possible to make it very useful on say, a 24 px vertical panel. 

Three things could help though: shrinking instead of cutting the icon, being able to remove the units or place the units label under the numbers,  and maybe ellipsizing the label text so that the least significant figure would always be the first to go, even with the panel on the right side of the screen.

**Comment by pF-arQon** [comment]
Yeah, there are multiple issues with the code, though the icon problems are minor compared to the applet being literally nonfunctional since the 1.26 regression.

Like I say, all the bugs I've seen so far are fixed in the patches I posted six weeks ago, including the icon: they just need someone to actually merge them. There's not much point discussing cosmetic changes while even the most basic pieces are still broken, because the situation will be different once those are working.

**Comment by lukefromdc** [comment]
patches posted where? I don't see a PR concerning this

**Comment by pF-arQon** [comment]
In the first post:

> https://bugs.launchpad.net/ubuntu/+source/mate-applets/+bug/1841854 , which contains fixes for both this and the icon positioning bug

There are more bugs with it - as you say, it's just not very good at coping with spacing/orientation - but my time is limited so I had to prioritize making it functional first.

**Comment by lukefromdc** [comment]
We cannot test and merge a PR from launchpad etc w/o manually cutting and pasting it into a PR ourselves. If you could do that we could then much more easily test it
